src/shogun/lib/SGSparseMatrix.h

```diff
@@ -107,15 +107,15 @@ template <class T> class SGSparseMatrix : public SGReferencedData
 		{
 			REQUIRE(i_row>=0, "index %d negative!\n", i_row);
 			REQUIRE(i_col>=0, "index %d negative!\n", i_col);
-			REQUIRE(i_row<num_vectors, "index should be less than %d, %d provided!\n",
-				num_vectors, i_row);
-			REQUIRE(i_col<num_features, "index should be less than %d, %d provided!\n",
-				num_features, i_col);
+			REQUIRE(i_row<num_features, "index should be less than %d, %d provided!\n",
```

Comment:
I know it is very minor. But could we do
"Provided index (%d) is larger than number of rows (%d)\n"

Capital letters at beginning of sentences, uniform style across Shogun. This is what our users see and it should be as clean as possible.